File: clients/swift/src/Exceptions/RateLimitedError.swift
import Foundation

public final class RateLimitedError: ApiError, @unchecked Sendable {
    public let states: [RateLimitState]

    public init(problem: Problem, states: [RateLimitState]) {
        self.states = states
        super.init(status: 429, problem: problem)
    }
}

File: clients/typescript/src/exceptions/RateLimitedError.ts
import type { Problem } from "../models/Problem.js";
import type { RateLimitState } from "../ratelimiting/RateLimitState.js";
import { ApiError } from "./ApiError.js";

export class RateLimitedError extends ApiError {
  constructor(
    problem: Problem,
    public readonly states: RateLimitState[],
  ) {
    super(429, problem);
    this.name = "RateLimitedError";
  }
}
